src/Microsoft.AspNet.Http/Features/FormFeature.cs

```diff
@@ -63,7 +65,15 @@ public bool HasFormContentType
             }
         }
 
-        public IFormCollection Form { get; set; }
+        public IFormCollection Form
+        {
+            get { return _form; }
+            set
+            {
+                _parsedFormTask = null;
```

Comment:
You've got the non-async method; so didn't want to set the task if that's the only path used